src/logistration/RedirectLogistration.jsx

```diff
@@ -1,22 +1,33 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import { getConfig } from '@edx/frontend-platform';
+
 function RedirectLogistration(props) {
-  const { success, redirectUrl } = props;
+  const { finishAuthUrl, success, redirectUrl } = props;
+
   if (success) {
-    window.location.href = redirectUrl;
+    // If we're in a third party auth pipeline, we must complete the pipeline
+    // once user has successfully logged in. Otherwise, redirect to the specified redirect url.
+    if (finishAuthUrl) {
+      window.location.href = getConfig().LMS_BASE_URL + finishAuthUrl;
```

Comment:
I think it should be independent of success. Because success would always false in your case because success is being passed from loginResult props.

Reply:
So basically, if the social account is associated with edX account, the pipeline will initiate the redirection to dashboard and we will never return back to MFE. 
Another scenario where above redirection is needed is when user doesn't have an edX account associated with social auth. In this case we come back to MFE in the middle of the running pipeline. We ask user to enter the usual email and password to his account and press "sign in". Once user clicks sign in `login_ajax` is called and we get loginResult back if the auth was successful. Now in this case where we have a pipeline running and login is successful, we want to redirect back to pipeline i.e `finishAuthUrl`.
If we don't send it to `finishAuthUrl` it will not be able to link social auth with edX account